lib/classes.rb

```diff
@@ -0,0 +1,130 @@
+class Player
+  attr_reader :name
+  attr_accessor :piece
+  attr_accessor :score
+  def initialize(name, piece = ' ', score = 0)
+    @name = name
+    @score = score
+    @piece = piece
+  end
+
+  def change_score
+    @score += 1
+  end
+end
+
+class Match
+  SEPARATOR = '----------'.freeze
+
+  def initialize(matches = 0)
+    @hash = { 1 => 1, 2 => 2, 3 => 3, 4 => 4, 5 => 5, 6 => 6, 7 => 7, 8 => 8, 9 => 9 }
+    @turn = 0
+    @matches = matches
+  end
+
+  def who_is_first?(player_one, player_two)
+    value = %w[X O]
+    select = Random.new.rand(2)
+    if select.zero?
+      player_one.piece = value[0]
+      player_two.piece = value[1]
+      player_one.name
+    else
+      player_one.piece = value[1]
+      player_two.piece = value[0]
+      player_two.name
+    end
+  end
+
+  def display_board(board_hash)
+    board_hash['row_one'].call(@hash)
+    board_hash['separator'].call(SEPARATOR)
+    board_hash['row_two'].call(@hash)
+    board_hash['separator'].call(SEPARATOR)
+    board_hash['row_three'].call(@hash)
+  end
+
+  def place_move(player_one, puts_hash, board_hash)
+    move = 0
+    until ((1..9).include? move) && (@hash.value? move)
+      puts_hash['another_num'].call
+      move = gets.chomp
+      exit if move == 'exit'
+      move = move.to_i
+    end
+    @hash[move] = player_one.piece
+    display_board(board_hash)
+  end
```

Comment:
* [ ] No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

- [ ] The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing).([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/single_responsibility.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

- The responsibility of this method should be only placing a move on the board, instead, it has a loop that it's requesting for input and printing on screen, I think a great part of the behavior of this method also belongs in the main.rb file